recipes/hictk/build.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+export CONAN_NON_INTERACTIVE=1
+
+export CMAKE_BUILD_PARALLEL_LEVEL=${CPU_COUNT}
+export CTEST_PARALLEL_LEVEL=${CPU_COUNT}
+
+if [[ ${DEBUG_C} == yes ]]; then
+  CMAKE_BUILD_TYPE=Debug
+else
+  CMAKE_BUILD_TYPE=Release
+fi
+
+scratch=$(mktemp -d)
+export CONAN_HOME="$scratch/conan"
+
+# shellcheck disable=SC2064
+trap "rm -rf '$scratch'" EXIT
+
+# Conan doesn't detect compiler name and version when using cc/c++
+TMPBIN="$scratch/bin"
+mkdir "$TMPBIN"
+
+declare -a CMAKE_PLATFORM_FLAGS
+if [[ ${HOST} =~ .*darwin.* ]]; then
+  export MACOSX_DEPLOYMENT_TARGET=10.15  # Required to use std::filesystem
+  CMAKE_PLATFORM_FLAGS+=(-DCMAKE_OSX_SYSROOT="${CONDA_BUILD_SYSROOT}")
+  conan_profile='clang'
+else
+  CMAKE_PLATFORM_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE="${RECIPE_DIR}/cross-linux.cmake")
+
+  ln -sf "$CC" "$TMPBIN/gcc"
+  ln -sf "$CXX" "$TMPBIN/g++"
+
+  export PATH="$TMPBIN:$PATH"
+  export CC="$TMPBIN/gcc"
+  export CXX="$TMPBIN/g++"
+  conan_profile='gcc'
```

Comment:
I think this should not be necessary and the environment variables will be populated by conda during the build process

Reply:
This workaround is required to build boost with Conan using conda-forge compilers (more specifically b2, one of boost build deps).
Long story short, b2 is not configured correctly if the compiler does not have its "canonical" name.

Anyways, the workaround is no longer necessary now that boost is installed by conda.

Reply:
Having boost in the dependencies is definitely a better solution than compiling it like this. Just make sure that it also actually works :)